Added tests for autoselecting options and chart builder configuration saving

cryogenian opened the pull request:
@jonsterling @beckyconning  Please, review

jonsterling commented on test/src/Test/Selenium/Notebook/Getters.purs:
```diff
@@ -401,7 +400,7 @@ getCurrentChartOptions =
 
 getCurrentEditorChild :: String -> Check Element
-getCurrentEditorChild sel = do
+getCurrentEditorChild sel = retry 10 do
```
At first glance, this looks like way to introduce non-determinism—I don't know if it is avoidable since I'm not familiar with the reasons for its presence, but I thought I'd mention it in case anyone has some insight here.

beckyconning replied:
Yeah there probably is a really good reason for this but I'm not sure what it is. Why not a timeout based retry like `await`?

cryogenian replied:
Huh, I forgot about `waiter` it's the one I need here

beckyconning commented on test/config.json:
```diff
@@ -348,6 +349,32 @@
+        },
+        "flatVizOneOption": {
+            "pie": {
+                "measureOne": [".measureOne"],
```
It might be worth adding the suffix "Selector" to delineate between DOM selectors and other test values.

`measureOneSelector`

beckyconning commented on test/src/Test/Selenium/Notebook/Viz.purs:
```diff
@@ -24,9 +24,9 @@ import Data.List (length, List(..), fromList, filter, null, (!!))
 import Data.Maybe (isJust, isNothing, Maybe(..), maybe, fromMaybe)
 import Data.Maybe.Unsafe (fromJust)
 import Data.Traversable (traverse)
+import Data.Foldable (foldl)
```
:+1:

cryogenian commented:
* Replaced `retry 10` with `waiter`, added `waiter` to similar function. 
* Renamed `viz` to `vizSelectors`. (Now, we can definitely know, that it's selector i.e. 
```purescript
getConfig >>= _.vizSelectors >>> _.measure 
-- Or
do config <- getConfig 
     pure config.vizSelectors.getConfig 
``` 
) @beckyconning Is it ok?

jonsterling commented:
Looks good to me; I'll leave the final review/thumbs up to @beckyconning.
